### PR TITLE
Patch to let UHC make offline token requests to KeyCloak not gobble up the data

### PR DESCRIPTION
**Pull request by iphands**
To use this you can
```javascript
window.insights.chrome.auth.doOffline()
```

Then after the postback
```javascript
window.insights.chrome.auth.getPostbackUrl()
```

Note you can only call getPostbackUrl and obtain the data once

### CODE REVIEW

**Comment by elad661** [comment]
I tested this locally and it works as expected

**Comment by elad661** [comment]
To make the http request to get the token, I need a bit more info than the postback URL - I need the keycloak server address and the keycloak client_id. Can you add these to the API as well?

**Comment by iphands** [comment]
@karelhala, @elad661 provided a patch to do the actual POST needed to get the final offline token
I went ahead and refactored a bit, then added some unit tests. Can you please review and contribute any style changes you'd like to see? I want to get this merged quickly.

@elad661 please test the change as it is now and let me know if something functional is broken. Thanks.

File: src/js/jwt/jwt.js [hunk 1]
```diff
@@ -72,13 +73,33 @@ priv.decodeToken = (str) => {
...
     return str;
 };
 
+pub.doOffline = (key, val) => {
+    const url = urijs(window.location.href);
+    url.removeSearch(key);
+    url.addSearch(key, val);
+
+    const options = {
+        realm: 'redhat-external',
```
Comment (karelhala) [inline src/js/jwt/jwt.js]: Can you export [src/js/auth.js#L7~options](https://github.com/RedHatInsights/insights-chrome/blob/bc0725494efc0adc38059dd6bd19a7911d17de54/src/js/auth.js#L7) const and use it here as well? So we don't have to keep track of the same vars in two places.

File: src/js/jwt/insights/offline.test.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    describe('getOfflineToken', () => {
+        test('fails when there is no offline postbackUrl', async () => {
+            try {
+                await offline.getOfflineToken();
+            } catch (e) {
+                expect(e).toBe('not available');
+            }
+        });
+
+        test('POSTs to /token with the right parameters when input is good', () => {
```
Comment (karelhala) [inline src/js/jwt/insights/offline.test.js]: Missing async.

**Comment by elad661** [comment]
@iphands thanks, I tested this and it seems to work well.

**Comment by oourfali** [comment]
@elad661 any change needed on our side to have that fixed? as I still see network error in the UI after a few minutes.

**Comment by elad661** [comment]
@oourfali this is unrelated to the issue you're seeing. This PR is to enable the "offline API tokens" page. The issue you're seeing is a different one.

**Comment by oourfali** [comment]
I see.
Thanks.

**Pull request merged**